startreedata/thirdeye
# refactor(ui): TE-1371 re-architect subscription group components

#### Issue(s)

https://cortexdata.atlassian.net/browse/TE-1371

#### Description
Re-architect subscription group components similar to guided alerts creation flow

#### Screenshots

No changes except url
<img width="1479" alt="Screenshot 2023-04-10 at 6 15 02 PM" src="https://user-images.githubusercontent.com/3389485/231030158-6b019ee8-a2ad-4ffc-84a3-80dde89b817d.png">
<img width="1466" alt="Screenshot 2023-04-10 at 6 15 11 PM" src="https://user-images.githubusercontent.com/3389485/231030161-9bb2cae8-2154-4897-8fe5-2f1f131dd3cf.png">

#### How to test
Scenarios tested:
1. create new subscription group with empty description name, save button is grayed out
2. create new subscription group without alerts & dimensions, a dialog is popped out asking for confirmation
3. input data is saved when going back and forth within the creation flow
4. created subscriptions can be shown correctly in view page
5. repeat the same tests for edit flow - the save button should be update instead.
6. edited subscription group should be saved.
